File: duplicate-zeros/duplicate-zeros.py
class Solution(object):
    def duplicateZeros(self, arr):
        """
        :type arr: List[int]
        :rtype: None Do not return anything, modify arr in-place instead.
        """
        l = len(arr)
        idx = 0             # ans's index
        while idx < l:
            if arr[idx] == 0:
                arr.insert(idx, 0)
                arr.pop()
                idx += 2
            else:
                idx += 1
        
        return arr

File: duplicate-zeros/duplicate-zeros.java
class Solution {
    public void duplicateZeros(int[] arr) {
        int zeroCnt = 0;
        for(int i = 0; i < arr.length; i++)
            if(arr[i] == 0)
                zeroCnt++;
        int[] ans = new int[arr.length + zeroCnt];
        int idx = 0;    // ans index
        
        for(int i = 0; i < arr.length; i++) {
            if(arr[i] != 0) {
                ans[idx] = arr[i];
                idx++;
            } else {
                ans[idx] = 0;
                ans[idx + 1] = 0;
                idx += 2;
            }
        }
        for(int i = 0; i < arr.length; i++)
            arr[i] = ans[i];
    }
}